InputValidation.php

```diff
@@ -0,0 +1,217 @@
+<?php
+declare(strict_types=1);
+
+
+/**
+ * @param array $options
+ * @return array
+ */
+function validateSelect(array $options): array
+{
+    if (!isset($options['select'])) {
+        return ['Missing  --select  option. Format: --select column1,column2,...'];
+    }
+
+    if ($options['select'] === '*') {
+        return [];
+    }
+
+    $pattern = '/[^\w,]/u';
+    if (preg_match($pattern, $options['select'])) {
+        return ['Invalid format for option  --select. Format: --select column1,column2,...'];
+    }
+
+    return [];
+}
+
+/**
+ * @param array $options
+ * @return array
+ */
+function validateFrom(array $options): array
+{
+    if (!isset($options['from'])) {
+        return ['Missing  --from  option. Format: --from filename' . FILE_EXTENSION];
+    }
+
+    if (substr($options['from'], -4) !== FILE_EXTENSION) {
+        return ['Invalid format: --from  option requires a valid ' . FILE_EXTENSION . ' file'];
+    }
+
+    $pattern = '/([a-z\d*?A-Z\d*?\-_]*\\' . FILE_EXTENSION . '$)/u';
+    preg_match($pattern, $options['from'], $matches);
+    if ($matches[0] !== $options['from']) {
+        return ['Invalid format for option  --from. Format: --from filename' . FILE_EXTENSION];
+    }
+    if (!file_exists(FILES_DIRECTORY . $options['from'])) {
+        return ["File '{$options['from']}' not found. Try moving your file into /Files directory"];
+    }
+
+    return [];
+}
+
+/**
+ * @param array $options
+ * @return array
+ */
+function validateSort(array $options): array
+{
+    if (isset($options['sort-column']) || isset($options['sort-mode']) || isset($options['sort-direction'])) {
+        if (!isset($options['sort-column']) || !isset($options['sort-mode']) || !isset($options['sort-direction'])) {
+            return ['Sorting option requires each of the following to be specified:' . PHP_EOL
+                . '     --sort-column     column_name' . PHP_EOL
+                . '     --sort-mode       natural|alpha|numeric' . PHP_EOL
+                . '     --sort-direction  asc|desc'
+            ];
+
+        }
+    }
+    if (!isset($options['sort-column']) && !isset($options['sort-mode']) && !isset($options['sort-direction'])) {
```

Comment:
This should be the first condition in the function in order to simplify the logic.

Reply:
solved